chalk-ir/src/debug.rs

```diff
@@ -39,7 +39,7 @@ impl<I: Interner> Debug for Const<I> {
     }
 }
 
-impl<I: Interner> Debug for ConstEvaluated<I> {
+impl<I: Interner> Debug for ConcreteConst<I> {
```

Comment:
Hmm, Rustc calls this [`ConstKind::Value`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/enum.ConstKind.html#variant.Value), maybe we want to take that name

Reply:
I thought about naming it `ConstValue`, my main reasoning against it was that `ConstData` and `ConstValue` are too similar, `ConcreteConst` was one of the names eddyb proposed

Reply:
OK. I don't care too much which name we adopt, we'll have to reconcile them eventually. I can see why `ConstData` and `ConstValue` seem close -- I wonder if this would feel better if we used `ConstKind` instead of `ConstData` (and similarly `TyKind` instead of `TyData` etc)...I was thinking of adopting that rename on the basis of "seems about equivalent and it's easier to port chalk than rustc".

Reply:
Well, now we have `ConstValue` and `ConstData` and `ConcreteConst` all at once :upside_down_face: